src/main/scala/beam/utils/VMInfoCollector.scala

```diff
@@ -0,0 +1,57 @@
+package beam.utils
+
+import java.lang.management.ManagementFactory
+
+import javax.management.{
+  InstanceNotFoundException,
+  MBeanException,
+  MBeanServer,
+  MalformedObjectNameException,
+  ObjectName,
+  ReflectionException
+}
+
+object VMInfoCollector {
+  private val DIAGNOSTIC_COMMAND_MBEAN_OBJECT_NAME = "com.sun.management:type=DiagnosticCommand"
+
+  def apply(): VMInfoCollector = {
+    try {
+      val objectName = new ObjectName(DIAGNOSTIC_COMMAND_MBEAN_OBJECT_NAME)
+      new VMInfoCollector(objectName)
+    } catch {
+      case _: MalformedObjectNameException =>
+        throw new RuntimeException(
+          "Unable to create an ObjectName and so unable to create instance of VirtualMachineDiagnostics"
+        )
+    }
+  }
+}
+
+class VMInfoCollector(val objectName: ObjectName) {
+  private val server: MBeanServer = ManagementFactory.getPlatformMBeanServer
+
+  def gcRun(): Unit = invoke("gcRun")
+
+  def gcClassHistogram: String = {
+    gcRun()
+    invoke("gcClassHistogram")
+  }
+
+  //  commands examples:
```

Comment:
I'd suggest to use Scala type system to help with possible params. It can be done by using ADT, for example,
```scala
sealed trait JFRCommand {
 def asStr: String
}
object JFRCommand {
  case object GcRun extends JFRCommand  {
      def asStr: String = "gcRun"
  }
}
```

Reply:
ok, looks cool, thanks!